app/assets/javascripts/modules/moj.matchVisitors.js

```diff
@@ -0,0 +1,206 @@
+      this.totalVisitors = this.$el.find('select').length;
+    },
+
+    bindEvents: function() {
+      this.$el.find('select').on('change', $.proxy(this.changeSelect, this));
+      this.$el.find(this.notContactCheckbox).on('change', $.proxy(this.changeNotOnList, this));
+      this.$el.find(this.bannedCheckbox).on('change', $.proxy(this.changeBanned, this));
+    },
+
+    changeSelect: function(e) {
+      var el = e.currentTarget,
+        contactData = $(el).find(':selected').data('contact'),
+        adding = el.value == '0',
+        parent = this.findParent(el);
+
+      this.toggleSelectOptions(el);
+      this.toggleCheckbox(el, adding);
+      this.processVisitor(parent, !adding);
+      this.checkStatus();
+      if (this.isBanned(contactData)) {
+        this.setBanned(parent, this.isBanned(contactData));
+      }
+    },
+
+    changeNotOnList: function(e) {
+      var el = $(e.currentTarget),
+        parent = this.findParent(el),
+        isChecked = el.is(':checked');
+
+      this.processVisitor(parent, isChecked);
+      this.checkStatus();
+    },
+
+    changeBanned: function(e) {
+      var el = $(e.currentTarget),
+        parent = this.findParent(el),
+        isChecked = el.is(':checked');
+
+      this.setVisitorBanned(parent, isChecked);
+      this.checkStatus();
+    },
+
+    findParent: function(el) {
+      return $(el).parents('li');
+    },
+
+    processVisitor: function(el, processed) {
+      var select = false,
+        noContact = false;
+      select = $(el).find('select option:selected').val();
+      noContact = $(el).find(this.notContactCheckbox).is(':checked');
+
+      if (select == '0' && noContact == false) {
+        $(el).attr('data-processed', false);
+      } else {
+        $(el).attr('data-processed', true);
+      }
+    },
+
+    getProcessed: function() {
+      return this.$el.find('[data-processed="true"]').length;
+    },
+
+    setVisitorBanned: function(el, banned) {
+      $(el).attr('data-banned', banned);
+    },
+
+    isBanned: function(el) {
+      return $(el).attr('data-banned') == 'true';
+    },
+
+    checkStatus: function() {
+      this.checkAdultStatus();
+      this.checkTotalStatus();
+    },
+
+    checkAdultStatus: function() {
+      function isBigEnough(value) {
+        return function(element, index, array) {
+          return (element >= value);
+        }
+      }
+      var adultNumber = this.getAges().filter(isBigEnough(18));
+      var noAdults = adultNumber < 1;
+
+      if (noAdults && this.getProcessed() >= 1) {
+        this.showEl(this.$noAdultMessage);
+      } else {
+        this.hideEl(this.$noAdultMessage);
+      }
+    },
+
+    checkTotalStatus: function() {
+      var unprocessed = this.getProcessed() < this.totalVisitors;
+
+      if (unprocessed) {
+        this.showEl(this.$notAllMessage);
+      } else {
+        this.hideEl(this.$notAllMessage);
+      }
+    },
+
+    getAges: function() {
+      var self = this,
```

Comment:
self/this is quite common in JS and $.proxy is unnecessary in these small functions